src/service/config.rs

```diff
@@ -254,6 +260,74 @@ impl<T> ProtocolHandle<T> {
     }
 }
 
+/// Control whether the protocol handle method requires blocking to run
+/// default is 0b1111, all function use blocking
+///
+/// flag & 0b1000 > 0 means `connected` use blocking
+/// flag & 0b0100 > 0 means `disconnected` use blocking
+/// flag & 0b0010 > 0 means `received` use blocking
+/// flag & 0b0001 > 0 means `notify` use blocking
+#[derive(Copy, Clone, Debug)]
+pub struct ProtocolFlag(u8);
```

Comment:
suggest using `bitflags` crate

Reply:
Not planning to import this crate for the time being, the result of code generation is the same